src/main.ts

```diff
@@ -47,6 +47,35 @@ function ensureNamespace(namespace: string, objectId: string): string {
 	return `${namespace}.${objectId}`;
 }
 
+function getTypefromValue(value:any): ioBroker.CommonType|undefined {
+	switch (typeof(value)){
+		case "object":
+			// Handle typeof [] === "object"
+			if (Array.isArray(value)){
+				return "array";
+			// Handle typeof {} === "object"
+			}else if(Object.prototype.toString.call(value) === "[object Object]"){
+				return "object";
+			// typeof null === "object"
+			}
+		case "number":
```

Comment:
you should add a `return;` above this line or `null` will always fall through to the `number` case

Reply:
nevermind, you're not calling this with `null`. I keep getting confused by this.